db/migrations/tss-migrations/2024-02-15.0-add-stellar-signatories-table.sql

```diff
@@ -0,0 +1,18 @@
+-- +migrate Up
+
+CREATE TABLE stellar_signatories (
+    public_key VARCHAR(64) PRIMARY KEY,
+    encrypted_private_key VARCHAR(256),
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
```

Comment:
I'm still debating if this is the best name for this table. Here's a list of names I discarded:
- **stellar_keypairs**: keypair already has a solid meaning of the pair of public/private stellar keys, and it's used inside the signature clients. Also, this table would have the private key encrypted, which is a bit different than the traditional keypair.
- **distribution_accounts**: this clearly means that this table will be used explicitly for distribution accounts, which may not be true in the mid-term. For instance, we could decide to store all private keys here, or at least store the host private key alongside the other distribution keys. For that reason, I wanted to use a more generic term.

The other option I have in mind is stellar_credentials, I'm keen to hear from others what they think would work best

Reply:
`stellar_signatories` is a good option. 

Another one would be `stellar_key_vault` ? there are a few products named `key vault` which do just that: store cryptographic keys. 
https://azure.microsoft.com/en-ca/products/key-vault
https://www.oracle.com/ca-en/security/database-security/key-vault/

Reply:
After some discussion offline we agreed to go with `vault` for the table name, and DBVault and DBVaultEntry for the object names.